core/state.h

```diff
@@ -43,7 +43,9 @@ struct Value {
         ARRAY = 0x10,
         FUNCTION = 0x11,
         OBJECT = 0x12,
-        STRING = 0x13
+        STRING = 0x13,
+
+        ANY = 0xFF00
```

Comment:
I am not sure whether it is acceptable to add this ENUM. I am open to suggestions though.